rollastic/node.py

```diff
@@ -54,13 +55,26 @@ def populate(self):
         self.clear()
 
         info = self.cluster.es.nodes.info(self.node_id)['nodes'].get(self.node_id, {})
+        isbad = False
```

Comment:
We can remove the variable entirely if we let the exception flow handle this for us. I suppose the downfall to this approach is that it doesn't allow us to miss collecting one or the other. We can fix that too by just checking for this at the end instead

```
try:
   info = self.cluster.es.nodes.info(self.node_id)['nodes'].get(self.node_id, {})
   self.update(info) 

   stats = self.cluster.es.nodes.stats(self.node_id)['nodes'].get(self.node_id, {})
   self.update(stats)

   if not info or not stats:
      raise Exception('Got bad result attempting to retrieve values. node_id={} info={} stats={}')...

   self.unsuccessful_retrieval_attempts = 0 # does this matter? once we update we're done with the node, right?
except Exception as e:
   self.unsuccessful_retrieval_attempts += 1
   if self.unsuccessful_retrieval_attempts > 10:
      raise Exception(''Reached retrieval limit for node...)```